JS/ViewControllers/settingsViewController.js

```diff
@@ -0,0 +1,429 @@
+                              </label>
+                            </p>
+                          </div>
+                        </div>
+                        <div class="row">
+                          <div class="col s4">
+                            <p>Translation</p>
+                          </div>
+                          <div class="col s8">
+                            <p>
+                              <label>
+                                <input type="checkbox" class="filled-in" name="translation" id="trans-check"/>
+                                <span></span>
+                              </label>
+                            </p>
+                          </div>
+                        </div>
+                      </div>
+                    </div>
+                  </div>
+                </div>
+            <div class="row">
+              <div class="col s12">
+                <div class="divider"></div>
+              </div>
+            </div>
+            <div class="row">
+              <div class="col s12 offset-l3 l6">
+                <div class="input-field col s12">
+                ` + langOptions + `
+                  <label>Paralel text</label>
+                </div>
+              </div>
+              <div class="col s12 center-align">
+                <a class="waves-effect waves-light btn-large" id="start-lecture"><i class="material-icons right">play_arrow</i>Play</a>
+              </div>
+            </div>
+        </div>
+        `;
+        super.renderHtml(htmlView);
+
+        var elems = document.querySelectorAll('select');
+        var instances = M.FormSelect.init(elems);
+    }
+
+    setupProperties() {
+
+        this.backBtn = $('#back-to-lecture-select');
+        this.startLectureBtn = $('#start-lecture');
+
+        this.playMode = document.getElementById('play-mode');
+        this.transLang = $('#translation-lang');
+
+        this.script = $('#script-check');
+        this.paralel = $('#trans-check');
+        this.direction = $('#direction');
+        this.pause = $('#pause');
+        this.repeat = $("#repeat");
+        this.pauseRepeat = $("#pause_repeat");
+    }
+
+    setupEventListeners() {
+        this.onBackBtnClick = this.onBackBtnClick.bind(this);
+        this.onStartLectureClick = this.onStartLectureClick.bind(this);
+
+        this.startLectureBtn.on('click', this.onStartLectureClick);
+        this.backBtn.on('click', this.onBackBtnClick);
+        this.playMode.onchange = () => {
+            this.setupSettingsOnPlayModeChange();
+        }
+    }
+
+    // After selecting the options for lecture this method shows the next page.
+    presentNextController() {
+        this.navigationController.present(new PlayViewController(this.lectureData));
+    }
+
+    // private methots
+
+    onBackBtnClick() {
+        this.navigationController.presentBefore();
+    }
+
+    onStartLectureClick() {
+        var el = document.getElementById('play-mode');
```

Comment:
Toto uz mas raz pretraverzovane a odchytene v premennej v `setupProperties` tuto sa to robi zbytocne znovu